usr/src/uts/common/fs/zfs/zcp_get.c

```diff
@@ -422,16 +422,21 @@ get_special_prop(lua_State *state, dsl_dataset_t *ds, const char *dsname,
 	case ZFS_PROP_INCONSISTENT:
 		numval = dsl_get_inconsistent(ds);
 		break;
-	case ZFS_PROP_RECEIVE_RESUME_TOKEN:
-		VERIFY3U(strlcpy(strval, get_receive_resume_stats_impl(ds),
-		    ZAP_MAXVALUELEN), <, ZAP_MAXVALUELEN);
+	case ZFS_PROP_RECEIVE_RESUME_TOKEN: {
+		char *token = get_receive_resume_stats_impl(ds);
+		VERIFY3U(strlcpy(strval, token, ZAP_MAXVALUELEN), <,
+		    ZAP_MAXVALUELEN);
+		strfree(token);
 		if (strcmp(strval, "") == 0) {
-			VERIFY3U(strlcpy(strval, get_child_receive_stats(ds),
-			    ZAP_MAXVALUELEN), <, ZAP_MAXVALUELEN);
+			token = get_child_receive_stats(ds);
+			VERIFY3U(strlcpy(strval, token, ZAP_MAXVALUELEN), <,
+			    ZAP_MAXVALUELEN);
+			strfree(token);
 			if (strcmp(strval, "") == 0)
 				error = ENOENT;
 		}
 		break;
+	}
```

Comment:
is it correct for '}' before `break` ?

Reply:
Yes, this is correct and generally how we structure it when you have a scope inside of the case statement.